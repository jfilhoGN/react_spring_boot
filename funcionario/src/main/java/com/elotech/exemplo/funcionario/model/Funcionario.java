package com.elotech.exemplo.funcionario.model;

import javax.persistence.*;

@Entity
public class Funcionario {

    @Id
    @GeneratedValue
    private Long id;

    @Column
    private String nome;

    @Column
    private String rg;

    @Column
    private String endereco;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getRg() {
        return rg;
    }

    public void setRg(String rg) {
        this.rg = rg;
    }

    public String getEndereco() {
        return endereco;
    }

    public void setEndereco(String endereco) {
        this.endereco = endereco;
    }

}
