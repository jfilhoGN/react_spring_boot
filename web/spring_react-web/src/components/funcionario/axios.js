import { ChangeSenhaFuncionario } from './changeSenhaFuncionario';

const POST_ENDPOINT = 'http://192.168.1.132:8080/api/colaboradores/updateSenha';

module.exports = {
  post: jest.fn((url) => {
    switch (url) {
      case POST_ENDPOINT:
        return Promise.resolve({ 
            "senhaAntiga":"12345",
            "senhaNova": "1234"
        },{
            headers:{
                'id':'28A5B171C9FF4FBFB4BC0481C0BD0435',
                'token':'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiIyOEE1QjE3MUM5RkY0RkJGQjRCQzA0ODFDMEJEMDQzNSJ9.T4K4br5kaEboh4UIl_gkhYi3FcMD1Wk_bIcQWVv-tRc'
            }  
        });
        
    }
  })
};