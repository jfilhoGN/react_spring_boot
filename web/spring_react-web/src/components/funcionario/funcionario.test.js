import React from 'react';
import {shallow} from 'enzyme';
import sinon from 'sinon';
import toJson from 'enzyme-to-json';
import Funcionario from './funcionario';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

describe('Funcionario', () => {

    test('render Funcionario ', () => {
        const history = {
            goBack: jest.fn(),
        }
        global.sessionStorage.getItem.mockReturnValue('{"id": "28A5B171C9FF4FBFB4BC0481C0BD0435", "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiIyOEE1QjE3MUM5RkY0RkJGQjRCQzA0ODFDMEJEMDQzNSJ9.T4K4br5kaEboh4UIl_gkhYi3FcMD1Wk_bIcQWVv-tRc"}');
        
        
        
        const mock = new MockAdapter(axios);
        mock.onGet('http://192.168.1.132:8080/api/colaboradores/me').reply(200,{headers:{
            'id': "28A5B171C9FF4FBFB4BC0481C0BD0435",
            'token': "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiIyOEE1QjE3MUM5RkY0RkJGQjRCQzA0ODFDMEJEMDQzNSJ9.T4K4br5kaEboh4UIl_gkhYi3FcMD1Wk_bIcQWVv-tRc"
        }});
        const wrapper = shallow(<Funcionario history={history} />);
        expect(wrapper).toMatchSnapshot();    
    });

    

    test('values of state', () => {
        /* const history = {
            goBack: jest.fn(),
        }
        global.sessionStorage.getItem.mockReturnValue('{"id": "28A", "token":"eyJ0eX"}')
        const wrapper = shallow(<Funcionario history={history} />); */
    });

    test('getFuncionario()', () => {
        //arrumar
    });

    test('componentDidMount()', () => {
        //arrumar
    });

    test('handleUpdate', () => {
        //arrumar
    });

    /* test('render component', () => {
        const history = {
            goBack: jest.fn()
        }
        const wrapper = shallow(<Funcionario history={history}/>);
        const component = wrapper.dive();
        expect(toJson(component)).toMatchSnapshot();
    });


    test('values of state', () => {
        const history = {
            goBack: jest.fn()
        }
        const wrapper = shallow(<Funcionario  history={history}/>)
        expect(wrapper.state()).toMatchSnapshot();

        //expect(history.goBack).toHaveBeenCalledTimes(1);
    }); */

});