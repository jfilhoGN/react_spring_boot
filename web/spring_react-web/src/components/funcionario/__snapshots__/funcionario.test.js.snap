// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Funcionario render Funcionario  1`] = `
<Card>
  <Row>
    <Col>
      <form
        className="col s12"
      >
        <Input
          checked={false}
          id="nome"
          label="Nome"
          name="nome"
          onChange={[Function]}
          placeholder="teste"
          s={12}
          type="text"
          value=""
        >
          <Icon
            small={true}
          >
            person
          </Icon>
        </Input>
        <Input
          checked={false}
          id="email"
          label="Email"
          name="email"
          onChange={[Function]}
          placeholder="teste"
          s={12}
          type="text"
          value=""
        >
          <Icon
            small={true}
          >
            email
          </Icon>
        </Input>
        <Input
          checked={false}
          id="cpf"
          label="CPF"
          name="cpf"
          onChange={[Function]}
          placeholder="teste"
          s={12}
          type="text"
          value=""
        >
          <Icon
            small={true}
          >
            assignment_ind
          </Icon>
        </Input>
        <Button
          className="grey darken-3 btn-small"
          name="action"
          node="button"
          onClick={[Function]}
          title="atualizar cadastro"
          type="submit"
        >
          <i
            className="material-icons"
          >
            update
          </i>
        </Button>
        <Button
          className="button-espaco grey darken-3 btn-small"
          name="action"
          node="button"
          onClick={[Function]}
          title="mudar senha"
          type="submit"
        >
          <i
            className="material-icons"
          >
            lock_open
          </i>
        </Button>
      </form>
    </Col>
  </Row>
  <Row>
    <h4>
      <center>
        Tarefas
      </center>
    </h4>
    <center>
      <p>
        <b>
          Para Realizar: 
        </b>
         
         
      </p>
    </center>
    <center>
      <p>
        <b>
          Realizando: 
        </b>
         
         
      </p>
    </center>
    <center>
      <p>
        <b>
          em Teste: 
        </b>
         
      </p>
    </center>
    <center>
      <p>
        <b>
          Concluido: 
        </b>
         
      </p>
    </center>
  </Row>
</Card>
`;
